### DISCUSSION TITLE
how argocd knows which namespace to find tiller

### DISCUSSION THREAD

**Discussion by ishallbethat** [argoproj/argo-cd#5361] [Q&A]
Hi guys
I'm so curious how argocd knows where to find tiller.
I set up helm application but i don't find anywhere in configuration to specify where tiller is.
How does it know ?

**Comment by timstoop** [comment]
Tiller is only deployed for Helm v2 and is deprecated, so you really shouldn't use it anymore. You should really consider switching to Helm v3.

That aside, the default location of Tiller is in `kube-system`, so it assumes that Tiller is running there.

**Comment by jannfis** [comment]
Hi @IamGabrielWu, Argo CD does not use or require tiller for Helm v2. It uses `helm template` to render the manifests, and then reconciles them into the cluster.

**Comment by ishallbethat** [comment]
I know helm V2 is deprecated and we should consider using Helm v3. But it took time for our SRE team to migrate. 
As mentioned by @jannfis "Argo CD does not use or require tiller for Helm v2. It uses helm template to render the manifests, and then reconciles them into the cluster."   My concern is our tiller is in namespace "helm", not "kube-system" and "directly reconciles manifests into the cluster" I think this will bring inconsistence between release data in tiller database and the reality @timstoop 

In argocd UI, i do not see any place to configure tiller namespace.
![image](https://user-images.githubusercontent.com/11718739/106533305-d74c4b80-652c-11eb-9556-c258148f5026.png)

**Comment by ishallbethat** [comment]
see the difference between helm v2 and helm v3 https://helm.sh/docs/faq/#improved-upgrade-strategy-3-way-strategic-merge-patches

**Comment by timstoop** [comment]
@IamGabrielWu No, @jannfis is right, Helm from ArgoCD does not use Tiller at all. It only generates the manifests and deploys them that way, not storing any state in the cluster. So assuming you do not mix and match manual maintained releases and automated-with-argo releases, you should be fine. There is not reconciliation being done. (At least not via Helm artifacts on your cluster.)
